### PR TITLE
Automatic backport

### LINKED ISSUE #29: Allow to automatically backport merged pull request with label

### PR DESCRIPTION
**Pull request by sileht**
Closes #29

### CODE REVIEW

**Comment by sileht** [comment]
Pull-request updated, HEAD is now d43519b3cb559ae58609a489770ed3387ba98043

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 69254aea197b03464018f39557819182ddf2343a

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 0fef228b98b1d0017448e881eca83664c6238d7e

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 736866de6ed0975f468e5105d1e4610248849ec1

File: doc/source/configuration.rst [hunk 1]
```diff
@@ -54,6 +54,14 @@ A ``Rule`` is a dictionary with the following optional keys:
    * - ``merge_strategy``
      - :ref:`Merge Strategy`
      - The method to use to merge pull requests.
+   * - ``automated_backport_labels``
+     - ``dict``
+     - A dictionary where key ares `labels
```
Comment (jd) [inline doc/source/configuration.rst]: keys

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 7ad86b76c52a55521c9051fdf495e933b11dea10

File: mergify_engine/backports.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                status = git("status").decode("utf8")
+                git("add", "*")
+                git("commit", "-a", "--no-edit")
+
+                body += ("\n\nCherry-pick of %s have failed:\n```\n%s```\n\n"
+                         % (commit.sha, status))
+
+        git("push", "origin", bp_branch)
+    except Exception:
+        LOG.exception("backport failure")
```
Comment (jd) [inline mergify_engine/backports.py]: not a fan of log.exception, rather log.<level> exc_info=True, you also lack contetxt info here I guess.

File: mergify_engine/backports.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        return
+    finally:
+        git.cleanup()
+
+    if cherry_pick_fail:
+        body += (
+            "To fixup this pull request, you can check out it locally. "
+            "See documenation: "
```
Comment (jd) [inline mergify_engine/backports.py]: documentation

**Comment by sileht** [comment]
Pull-request updated, HEAD is now ad93b6dd269542ca3145a98bce38d97aaf40149e

**Pull request merged**